client/extensions/woocommerce/app/dashboard/pre-setup-view.js

```diff
@@ -63,6 +63,13 @@ class PreSetupView extends Component {
 	onChange = ( event ) => {
 		const address = this.state.address;
 		address[ event.target.name ] = event.target.value;
+
+		// Did they change the country? Force an appropriate state default
+		if ( 'country' === name ) {
```

Comment:
Fixed in https://github.com/Automattic/wp-calypso/pull/15541/commits/17f50964f1f6156641032d53a6b96785ad2895a0